src/coreclr/jit/lclmorph.cpp

```diff
@@ -678,16 +678,52 @@ class LocalAddressVisitor final : public GenTreeVisitor<LocalAddressVisitor>
 
         // In general we don't know how an exposed struct field address will be used - it may be used to
         // access only that specific field or it may be used to access other fields in the same struct
-        // be using pointer/ref arithmetic. It seems reasonable to make an exception for the "this" arg
-        // of calls - it would be highly unsual for a struct member method to attempt to access memory
+        // by using pointer/ref arithmetic. It seems reasonable to make an exception for the "this" arg
+        // of calls - it would be highly unusual for a struct member method to attempt to access memory
         // beyond "this" instance. And calling struct member methods is common enough that attempting to
         // mark the entire struct as address exposed results in CQ regressions.
-        bool isThisArg = user->IsCall() && (user->AsCall()->gtCallThisArg != nullptr) &&
-                         (val.Node() == user->AsCall()->gtCallThisArg->GetNode());
+        GenTreeCall* callTree  = user->IsCall() ? user->AsCall() : nullptr;
+        bool         isThisArg = (callTree != nullptr) && (callTree->gtCallThisArg != nullptr) &&
+                         (val.Node() == callTree->gtCallThisArg->GetNode());
         bool exposeParentLcl = varDsc->lvIsStructField && !isThisArg;
 
-        m_compiler->lvaSetVarAddrExposed(exposeParentLcl ? varDsc->lvParentLcl
-                                                         : val.LclNum() DEBUGARG(AddressExposedReason::ESCAPE_ADDRESS));
+        bool hasHiddenStructArg = false;
+        if (m_compiler->opts.compJitOptimizeStructHiddenBuffer)
+        {
+            if (varTypeIsStruct(varDsc) && varDsc->lvIsTemp)
```

Comment:
Why check `lvIsTemp`? The meaning of that is not really well-defined.

As-is it means we're pessimizing user local (unnecessarily I would think).

Reply:
Sure, sounds reasonable. I will validate to see if it would be an easy change to remove it. If not, I will reserve it for next PR.